trace/opentracing.go

```diff
@@ -549,24 +565,33 @@ func (t Tracer) Extract(format interface{}, carrier interface{}) (ctx opentracin
 	}
 
 	if tm, ok := carrier.(opentracing.TextMapReader); ok {
-
 		// carrier is guaranteed to be an opentracing.TextMapReader by contract
 		// TODO support other TextMapReader implementations
-		traceID, err := strconv.ParseInt(textMapReaderGet(tm, TraceIDHeader), 10, 64)
-		spanID, err2 := strconv.ParseInt(textMapReaderGet(tm, SpanIDHeader), 10, 64)
-		parentID, err3 := strconv.ParseInt(textMapReaderGet(tm, ParentIDHeader), 10, 64)
-		if !(err == nil && err2 == nil && err3 == nil) {
+		parsedHeaders := false
+		var traceID int64
+		var spanID int64
+		for _, headers := range HeaderFormats {
+			var err1 error
+			var err2 error
+			traceID, err1 = strconv.ParseInt(textMapReaderGet(tm, headers.TraceID), 10, 64)
+			spanID, err2 = strconv.ParseInt(textMapReaderGet(tm, headers.SpanID), 10, 64)
+
+			if err1 == nil && err2 == nil {
+				parsedHeaders = true
+				break
+			}
+		}
+		if !parsedHeaders {
```

Comment:
Is this logically equivalent to 

```go
if traceID == 0 && spanID == 0 {
```

?

If so, I'd prefer to use that rather than another variable to keep track of, especially because it's more explicit about the end contract we're expecting.

Reply:
I think you mean

`if traceID != 0 && spanID != 0`

But yes, if `textMapReaderGet` fails to find a key, it will return the empty string. Calling `ParseInt` on the empty string will return 0, error.

Is it ever possible for the traceID or spanID to have been set, but for the value to actually be 0?

Reply:
I suppose there's a 1 in 2^63 chance, but it makes much more sense to 'return a failure if there was an error' than to 'assume there was an error if the id is a [technically valid] sentinel value' so I'm a bit bearish on changing the behavior as suggested here

Reply:
I don't think so - I believe `traceID == 0 && spanID == 0` is logically equivalent to `!parsedHeaders`. If we successfully parse headers, we expect `parsedHeaders` to be `true`, and we expect both IDs to be non-zero.

> Is it ever possible for the traceID or spanID to have been set, but for the value to actually be 0?


I *guess* someone could set both headers to zero, but it'd be meaningless in the context of tracing. I'm comfortable treating that as equivalent to the headers being empty.

Reply:
Unless that value is explicitly illegal, I'm bearish on this overall. "I was explicitly told there was an error" trumps "I think there was probably an error because reasons" for me every time ¯\_(ツ)_/¯

Reply:
We're going to retreat the result the same either way (ie, silently ignore the error). And it's a pretty Go pattern to design around the zero value for types such that they're meaningful. In this case, the meaning is that no semantically valid value was extracted using any of the format types defined.

Reply:
Oh geez, it took me like an entire month, but I just finally parsed your first comment in this thread correctly.

I thought we were all looking at line 582 (inside the for loop), and didn't realize this comment was about the `if !parsedHeaders` at the end. I'm going to make the right change now.